sql/set_var.cc

```diff
@@ -1013,7 +1013,7 @@ int set_var::check(THD *thd) {
     my_error(ER_WRONG_TYPE_FOR_VAR, MYF(0), var->name.str);
     return -1;
   }
-  int ret = (type != OPT_PERSIST_ONLY && var->check(thd, this)) ? -1 : 0;
+  int ret = var->check(thd, this) ? -1 : 0;
```

Comment:
var->check() calls either the  sys_var::check() or the plugin_var::check() ?

Reply:
Here sys_var::check() is called and it in turn will decide what to call - sys_var_pluginvar::do_check() or do_check() for corresponding sys var.